python/lsst/analysis/tools/tasks/makeMetricTable.py

```diff
@@ -0,0 +1,203 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = (
+    "MakeMetricTableConfig",
+    "MakeMetricTableTask",
+)
+
+
+import lsst.pipe.base as pipeBase
+import numpy as np
+from astropy.table import Table
+from lsst.pipe.base import connectionTypes as ct
+from lsst.skymap import BaseSkyMap
+
+from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+
+class MakeMetricTableConnections(
+    AnalysisBaseConnections,
+    dimensions=("skymap",),
+    defaultTemplates={"metricBundleName": "objectTableCore_metrics"},
+):
+    data = ct.Input(
+        doc="Metric bundle to read from the butler",
+        name="{metricBundleName}",
+        storageClass="MetricMeasurementBundle",
+        deferLoad=True,
+        dimensions=(
+            "skymap",
+            "tract",
+        ),
+        multiple=True,
+    )
+
+    skymap = ct.Input(
+        doc="The skymap that covers the tract that the data is from.",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    metricTable = ct.Output(
+        doc="A summary table of all metrics by tract.",
+        name="{metricBundleName}Table",
+        storageClass="ArrowAstropy",
+        dimensions=("skymap",),
+    )
+
+
+class MakeMetricTableConfig(AnalysisBaseConfig, pipelineConnections=MakeMetricTableConnections):
+    pass
+
+
+class MakeMetricTableTask(AnalysisPipelineTask):
+    """Turn metric bundles which are per tract into a
+    summary metric table.
+    """
+
+    ConfigClass = MakeMetricTableConfig
+    _DefaultName = "makeMetricTable"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        """Take a metric bundle for all the tracts and
+        seperate it into different metrics then make it
+        into a table.
+
+        Parameters
+        ----------
+        butlerQC : `lsst.pipe.base.QuantumContext`
+        inputRefs : `lsst.pipe.base.InputQuantizedConnection`
+        outputRefs : `lsst.pipe.base.OutputQuantizedConnection`
+        """
+
+        inputs = butlerQC.get(inputRefs)
+        skymap = inputs["skymap"]
+
+        # Make a list of tracts
+        tracts = []
+        for data in inputRefs.data:
+            tracts.append(data.dataId["tract"])
+
+        metricBundles = []
+        for inputHandle in inputs["data"]:
+            metricBundles.append(inputHandle.get())
+
+        outputs = self.run(tracts, metricBundles, skymap)
```

Comment:
Flyby review here. Could you move this data handling logic by overloading `_loadData` method and keep `runQuantum` the same from the base class? And keeping the signature of the `run` method the same as in the other tasks (i.e., `data`, `plotInfo` as arguments) would keep this in line with everything else. You might need to overload `parsePlotInfo` to return `None` to achieve that.

Reply:
I am happy with the way it is, either way stuff will be different from elsewhere and I tried various options and this seemed to work the best.